frontstage/controllers/case_controller.py

```diff
@@ -261,3 +194,30 @@ def validate_case_category(category):
         raise InvalidCaseCategory(category)
 
     logger.debug('Successfully validated case category', category=category)
+
+
+def get_cases_for_list_type_by_party_id(party_id, list_type='todo'):
+    logger.debug('Get cases for party for list', party_id=party_id, list_type=list_type)
+
+    business_cases = get_cases_by_party_id(party_id)
+    cases = filter_cases_by_case_group(business_cases)
+    history_statuses = ['COMPLETE', 'COMPLETEDBYPHONE', 'NOLONGERREQUIRED']
+    if list_type == 'history':
+        filtered_cases = [business_case
+                          for business_case in cases
+                          if business_case['caseGroup']['caseGroupStatus'] in history_statuses]
+    else:
+        filtered_cases = [business_case
+                          for business_case in cases
+                          if business_case['caseGroup']['caseGroupStatus'] not in history_statuses]
+
+    logger.debug("Successfully retrieved cases for party survey list", party_id=party_id, list_type=list_type)
+    return filtered_cases
+
+
+def filter_cases_by_case_group(cases):
+    logger.debug("Attempting to remove multiple cases for one case group")
```

Comment:
There's a bit of magic going on here that would be good to document in a docstring/comment as it isn't straight forward to scan.